scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -2983,16 +2985,114 @@ def __get_last_request_for_log(self, path):
 
         return scalyr_util.seconds_since_epoch(result)
 
-    def __create_log_config(self, parser, path, attributes, parse_format="raw"):
-        """Convenience function to create a log_config dict from the parameters"""
+    def __create_log_config(
+        self, info, k8s_info, container_attributes, parser, rename_logfile
+    ):
+        """
+        Creates a log_config from various attributes and then applies any `k8s_logs` configs that
+        might apply to this log
+        @return: A dict containing a log_config, or None if we couldn't create a valid config (i.e. log_path was empty)
+        """
 
-        return {
+        # Make sure we have a log_path for the log config
+        path = info.get("log_path", None)
+        if not path:
+            return None
+
+        # Build up the default config that we will use
+        result = {
             "parser": parser,
             "path": path,
-            "parse_format": parse_format,
-            "attributes": attributes,
+            "parse_format": self.__parse_format,
+            "attributes": container_attributes,
+            "rename_logfile": rename_logfile,
         }
 
+        # This is for a hack to prevent the original log file name from being added to the attributes.
+        if self.__use_v2_attributes and not self.__use_v1_and_v2_attributes:
+            result["rename_no_original"] = True
+
+        # If we don't have any k8s information then we don't match against k8s_log_configs
+        if k8s_info is None:
+            return result
+
+        # Now apply log configs
+        for i, config in enumerate(self.__k8s_log_configs):
```

Comment:
Ok, I would suggest breaking essentially this method out as a small abstraction for testing purposes.  I think its initialization should take container_attributes, parser, rename_logfile, and rename_no_original (either true or false).

Reply:
Unless we are instantiating this abstraction for every single container, then the only ones of these it can take are `rename_no_original`.

Everything other field will contain values specific to the container.

So, the abstraction I made takes the k8s_logs from the config, the logger, the log file parse format (json or cri) and the rename_no_original value (all of which are consistent across all containers).

There is then a `get_log_config` method that takes the remaining config specific items and returns the appropriate config.